sycl/source/device_selector.cpp

```diff
@@ -30,28 +30,37 @@ static bool isDeviceOfPreferredSyclBe(const device &Device) {
 
 device device_selector::select_device() const {
   vector_class<device> devices = device::get_devices();
-  int score = -1;
+  // SYCL specification uses -1 to reject a device from selection
```

Comment:
Perhaps on an example from the specification, but otherwise any negative value ends up in rejection.
So this comment is strange.

Reply:
Yeah that's true. I find the whole device selection scoring a bit strange, why negative values and not just 0 for rejected devices and any other score for valid devices? That way you could use an unsigned type which matches what you get back from other APIs or even OpenCL calls.
Fixed the comment in 6697fffaf80205b4d9c51fcb8a2b52acf046f24d

Reply:
Yes, That seems a good proposal. I do not remember how we got into the current situation. Perhaps it was done that way in Java? ;-)